hazelcast/src/test/java/com/hazelcast/util/SampleableConcurrentHashMapTest.java

```diff
@@ -44,4 +45,29 @@ public void samplesSuccessfullyRetrieved() {
         assertEquals(SAMPLE_COUNT, map.size());
     }
 
+    @Test
+    public void computeIfAbsentTest() {
+        final SampleableConcurrentHashMap<String, String> map =
+                new SampleableConcurrentHashMap<String, String>(10);
+
+        assertEquals(map.computeIfAbsent("key", new IFunction<String, String>() {
+            @Override
+            public String apply(String input) {
+                return "value";
+            }
+        }), "value");
+
+        for (int i = 0; i < 10; i++)
+            new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    assertEquals(map.computeIfAbsent("key", new IFunction<String, String>() {
```

Comment:
Problem here is, test won't fail if this assertion fails (because it's off-loaded to a new thread).